src/dialects/abstract/query-generator.js

```diff
@@ -2221,7 +2221,21 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
-    let fragment = `SELECT ${attributes.join(', ')} FROM ${tables}`;
+    let fragment = `SELECT`;
+
+    if (typeof options.maxExecutionTimeMs === 'number' && this.dialect.supports.maxExecutionTimeHint) {
+      const MINIMUM_EXECUTION_TIME_VALUE = 0;
+      const MAXIMUM_EXECUTION_TIME_VALUE = 4_294_967_295;
+
+      if (options.maxExecutionTimeMs >= MINIMUM_EXECUTION_TIME_VALUE
+        && options.maxExecutionTimeMs <= MAXIMUM_EXECUTION_TIME_VALUE) {
+        fragment += ` /*+ MAX_EXECUTION_TIME(${options.maxExecutionTimeMs}) */`;
+      } else {
+        throw new Error(`MAX_EXECUTION_TIME ms is must be between ${MINIMUM_EXECUTION_TIME_VALUE} and ${MAXIMUM_EXECUTION_TIME_VALUE}`);
```

Comment:
Inspect comes from the `node:util` package

```suggestion
        throw new Error(`maxExecutionTimeMs must be between ${MINIMUM_EXECUTION_TIME_VALUE} and ${MAXIMUM_EXECUTION_TIME_VALUE}, but it is ${inspect(options.maxExecutionTimeMs)}`);
```